internal/web/middleware.go

```diff
@@ -127,24 +131,40 @@ func AuthMW(log *logrus.Entry, tm token.Manager) Middleware {
 				parsedToken, err := tm.ParseWithClaims(tkn, JWTSigningSecret, &claims)
 				if err != nil {
 					w.WriteHeader(http.StatusUnauthorized)
-					fwd := forwardedHeader(r)
-					pluginID := normalizePluginID(fwd["by"])
+					fwd := ForwardedHeader(r)
+					pluginID := NormalizePluginID(fwd["by"])
+
+					// an empty plugin ID indicates an admin token
+					if pluginID == "" {
+						if err := JSONErrorResponse(w, err); err != nil {
+							log.WithError(err).Println("error parsing the token")
```

Comment:
This should print "sending json response".

Reply:
fixed